code/game/objects/items/melee/baton.dm

```diff
@@ -881,9 +881,7 @@
 	user.put_in_hands(brand_new_prod)
 
 /obj/item/melee/baton/security/cattleprod/can_baton(mob/living/target, mob/living/user)
-	if(!sparkler.activate())
-		return FALSE
-	return ..()
+	sparkler?.activate()
```

Comment:
Old code
```dm
/obj/item/melee/baton/security/cattleprod/baton_effect()
	if(!sparkler.activate())
		return BATON_ATTACK_DONE
	return ..()
```
Older code
```dm
/obj/item/weapon/melee/baton/cattleprod/baton_stun()
	if(sparkler.activate())
		..()
```
This is definitely working as designed. I don't know how this is a fix exactly